From 590c8dbebfddb71874de7a98e579264ebae6859f Mon Sep 17 00:00:00 2001
From: Martin Kletzander <mkletzan@redhat.com>
Date: Tue, 02 Apr 2013 11:51:28 +0000
Subject: Don't support sparse logical volumes

Sparse logical volumes are supported by libvirt, but only in case the
user takes care of the reallocation, which is not what virt-install
and virt-manager users want.  This patch defaults the logical volumes
to non-sparse ones and makes sure allocation is always equal to
capacity for such volumes.

Signed-off-by: Martin Kletzander <mkletzan@redhat.com>
---
diff --git a/virtinst/Storage.py b/virtinst/Storage.py
index 4d31266..1a39641 100644
--- a/virtinst/Storage.py
+++ b/virtinst/Storage.py
@@ -1,5 +1,5 @@
 #
-# Copyright 2008 Red Hat, Inc.
+# Copyright 2008, 2013 Red Hat, Inc.
 # Cole Robinson <crobinso@redhat.com>
 #
 # This program is free software; you can redistribute it and/or modify
@@ -1349,12 +1349,28 @@ class LogicalVolume(StorageVolume):
 
     def __init__(self, name, capacity, pool=None, pool_name=None, conn=None,
                  allocation=None, perms=None):
+        if allocation and allocation != capacity:
+            raise ValueError(_("Sparse logical volumes are not supported, "
+                               "allocation must be equal to capacity"))
         StorageVolume.__init__(self, name=name, pool=pool, pool_name=pool_name,
-                               allocation=allocation, capacity=capacity,
+                               allocation=capacity, capacity=capacity,
                                conn=conn)
         if perms:
             self.perms = perms
 
+    def set_capacity(self, capacity):
+        super(LogicalVolume, self).set_capacity(capacity)
+        self.allocation = capacity
+    capacity = property(StorageVolume.get_capacity, set_capacity)
+
+    def set_allocation(self, allocation):
+        if allocation != self.capacity:
+            raise ValueError(_("Sparse logical volumes are not supported, "
+                               "allocation must be equal to capacity"))
+        super(LogicalVolume, self).set_allocation(allocation)
+    capacity = property(StorageVolume.get_allocation, set_allocation)
+
+
     def _get_target_xml(self):
         return "%s" % self._get_perms_xml()
 
--
cgit v0.9.2
